ogr/ogrsf_frmts/geojson/ogrgeojsonwriter.cpp

```diff
@@ -871,10 +871,41 @@ json_object *OGRGeoJSONWriteAttributes(OGRFeature *poFeature,
             const size_t nLen = strlen(pszStr);
             poObjProp = nullptr;
             if (eSubType == OFSTJSON ||
-                (pszStr[0] == '{' && pszStr[nLen - 1] == '}') ||
-                (pszStr[0] == '[' && pszStr[nLen - 1] == ']'))
+                ((pszStr[0] == '{' && pszStr[nLen - 1] == '}') ||
+                 (pszStr[0] == '[' && pszStr[nLen - 1] == ']')))
```

Comment:
it doesn't look to me that this extra quoting is needed

Reply:
Yeah, I added it for clarity but no problems to remove it.